mne/channels/tests/test_standard_montage.py

```diff
@@ -73,3 +78,58 @@ def test_standard_superset():
     for key, value in m_1020._get_ch_pos().items():
         if key not in ('O10', 'O9'):
             assert_allclose(c_1005[key], value, atol=1e-4, err_msg=key)
+
+
+def _simulate_artinis_octamon():
+    """
+    Simulate artinis octamon channel data from numpy data.
+    This is to test data that is imported with missing or incorrect montage
+    info. This data can then be used to test the set_montage function.
+    """
+    data = np.random.normal(size=(16, 100))
+    ch_names = ['D1_S1 hbo', 'D1_S1 hbr', 'D1_S2 hbo', 'D1_S2 hbr',
+                'D1_S3 hbo', 'D1_S3 hbr', 'D1_S4 hbo', 'D1_S4 hbr',
+                'D2_S5 hbo', 'D2_S5 hbr', 'D2_S6 hbo', 'D2_S6 hbr',
+                'D2_S7 hbo', 'D2_S7 hbr', 'D2_S8 hbo', 'D2_S8 hbr']
+    ch_types = ['hbo', 'hbr', 'hbo', 'hbr',
+                'hbo', 'hbr', 'hbo', 'hbr',
+                'hbo', 'hbr', 'hbo', 'hbr',
+                'hbo', 'hbr', 'hbo', 'hbr']
+    sfreq = 10.  # Hz
+    info = create_info(ch_names=ch_names, ch_types=ch_types, sfreq=sfreq)
+    raw = RawArray(data, info, verbose=True)
+
+    return raw
+
+
+def test_artinis():
+    raw = _simulate_artinis_octamon()
+    old_info = raw.info.copy()
+    montage_octamon = make_standard_montage("artinis-octamon")
+    raw.set_montage(montage_octamon)
+    # First check that the montage was actually modified
+    assert_raises(AssertionError, assert_array_almost_equal,
+                  old_info["chs"][0]["loc"][:9],
+                  raw.info["chs"][0]["loc"][:9])
+    # Check a known location
+    assert_array_almost_equal(raw.info["chs"][0]["loc"][:3],
+                              [0.0616, 0.075398, 0.07347])
+    assert_array_almost_equal(raw.info["chs"][8]["loc"][:3],
+                              [-0.033875,  0.101276,  0.077291])
+    assert_array_almost_equal(raw.info["chs"][12]["loc"][:3],
+                              [-0.062749,  0.080417,  0.074884])
+    # fNIRS has two identical channel locations for each measurement
+    # The 10th element encodes the wavelength, so it will differ.
+    assert_array_almost_equal(raw.info["chs"][0]["loc"][:9],
+                              raw.info["chs"][1]["loc"][:9])
+    # Compare OctaMon and Brite23 to fsaverage
+    trans_octamon = compute_native_head_t(montage_octamon)
+    montage_brite = make_standard_montage("artinis-brite23")
+    trans_brite = compute_native_head_t(montage_brite)
+    fif = op.join(op.dirname(__file__), '..', '..', 'data', 'fsaverage',
+                  'fsaverage-trans.fif')
+    fsaverage = read_trans(fif)
+    assert_array_almost_equal(list(trans_octamon.values())[2],
+                              list(fsaverage.values())[2])
+    assert_array_almost_equal(list(trans_brite.values())[2],
+                              list(fsaverage.values())[2])
```

Comment:
@larsoner is that what you meant when you mentioned comparing to fsaverage? They are not the same however... Not sure how I should fix that.

Reply:
https://github.com/mne-tools/mne-python/pull/9141/checks?check_run_id=2156410557#step:12:3319